agent-connection/src/main/kotlin/com/luxoft/blockchainlab/corda/hyperledger/indy/PythonRefAgentConnection.kt

```diff
@@ -41,14 +66,24 @@ class PythonRefAgentConnection : AgentConnection {
 
     override fun getConnectionStatus(): AgentConnectionStatus = connectionStatus
 
-    private var counterParty: IndyParty? = null
-
     private lateinit var webSocket : AgentWebSocketClient
 
+    private val indyParties = HashMap<String, IndyPartyConnection>()
+
+    private fun getPubkeyFromInvite(invite: String): String {
+        val invEncoded = invite.split("?c_i=")[1]
+        val inv = Base64.getDecoder().decode(invEncoded).toString(Charsets.UTF_8)
+        val keyList = SerializationUtils.jSONToAny<Map<String, List<String>>>(inv)["recipientKeys"]
+        return keyList?.get(0)!!
+    }
+
+    /**
+     * Returns true, based on the "state" message returned from the agent, if the agent is already logged-in.
+     */
     private fun checkState(stateMessage: State?, userName: String) : Boolean {
```

Comment:
It seems the function checks isUserLoggedIn or not? Could we rename it, please

Reply:
renamed, please see 4d7a39afc27ef94281219b69b90ffedfad7154a3